From a5baaf44b75b0bd6d3411fc87531ddfd411e34fb Mon Sep 17 00:00:00 2001
From: Rogerio Pimentel <rogerio.pimentel@freescale.com>
Date: Thu, 8 Dec 2011 16:33:41 -0200
Subject: [PATCH] ENGR00169603 MX53 ARD: FlexCAN: Set lp_apm as clock source

The FlexCAN clock source must be lp_apm (24MHZ) instead
ipg_clock_root (60MHZ) to meet automotive clock requirements.

Signed-off-by: Rogerio Pimentel <rogerio.pimentel@freescale.com>
(cherry picked from commit b7456a4f5f6fa12235effbffe4e4d1b62159b948)
---
 arch/arm/mach-mx5/mx53_ard.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-mx5/mx53_ard.c b/arch/arm/mach-mx5/mx53_ard.c
index 378e1c1..b1252fb 100644
--- a/arch/arm/mach-mx5/mx53_ard.c
+++ b/arch/arm/mach-mx5/mx53_ard.c
@@ -455,6 +455,7 @@ static void flexcan_xcvr_enable(int id, int en)
 static struct flexcan_platform_data flexcan0_data = {
 	.core_reg = NULL,
 	.io_reg = NULL,
+	.root_clk_id = "lp_apm",
 	.xcvr_enable = flexcan_xcvr_enable,
 	.br_clksrc = 1,
 	.br_rjw = 2,
@@ -472,6 +473,7 @@ static struct flexcan_platform_data flexcan0_data = {
 static struct flexcan_platform_data flexcan1_data = {
 	.core_reg = NULL,
 	.io_reg = NULL,
+	.root_clk_id = "lp_apm",
 	.xcvr_enable = flexcan_xcvr_enable,
 	.br_clksrc = 1,
 	.br_rjw = 2,
-- 
1.5.4.4

